sn_client/src/uploader/upload.rs

```diff
@@ -120,6 +123,14 @@ pub(super) async fn start_upload(
             #[cfg(test)]
             trace!("UPLOADER STATE: finished uploading all items {uploader:?}");
 
+            if !uploader.max_repayments_reached.is_empty() {
```

Comment:
this is inside the check of `if uploader.all_upload_items.is_empty() {` ,
returning at this point shall get the statistics included in the UploadSummary (if needed), instead of return with the specific error

Reply:
Hey Qi, when the `max_repayments` is reached, the item is removed from `all_upload_items` and added `max_repayments_reached` struct. It is still an error, it is just that I chose to remove it from the flow there. We could've also had a filter. 
But it would still be an error, else people would get the "Successful uploaded message" and stuff, but in reality we failed to upload a couple items. Maybe we can also return the list of failed items inside ClientError instead.

Reply:
> Maybe we can also return the list of failed items inside ClientError instead.

Sounds good

Reply:
what I mean is that without the summary statistics, the user won't be able to know how many chunks uploaded successfully when received that error.

maybe, the error shall includes:
```
let upload_summary = UploadSummary { ... };
return Err(ClientError::MaximumRepaymentsReached {
                    failed_chunks: uploader.max_repayments_reached.into_iter().collect(),
                    upload_summary,
                });
```
then the user can clearly know how many succeeded, how many already presents, how much cost paid...